lib/jazzy/source_category.rb

```diff
@@ -0,0 +1,77 @@
+require 'jazzy/source_declaration'
+require 'jazzy/config'
+require 'jazzy/source_mark'
+require 'jazzy/jazzy_markdown'
+
+module Jazzy
+  # Category (group, contents) pages generated by jazzy
+  class SourceCategory < SourceDeclaration
+    extend Config::Mixin
+
+    def initialize(group, name, abstract, url_name, level = 1)
+      super()
+      self.type     = SourceDeclaration::Type.overview
+      self.name     = name
+      self.url_name = url_name
+      self.abstract = Markdown.render(abstract)
+      self.children = group
+      self.parameters = []
+      self.level      = level
+    end
+
+    # Group root-level docs into custom categories or by type
+    def self.group_docs(docs)
+      custom_categories, docs =
+        group_custom_categories(docs, config.custom_categories)
+      type_categories, uncategorized = group_type_categories(
+        docs, custom_categories.any? ? 'Other ' : ''
+      )
+      custom_categories + type_categories + uncategorized
+    end
+
+    def self.group_custom_categories(docs, categories, level = 1)
+      group = categories.map do |category|
+        children = category['children'].flat_map do |child|
+          if child.is_a?(Hash)
+            # Nested category, recurse
+            children, docs = group_custom_categories(docs, [child], level + 1)
```

Comment:
I renamed docs_with_name to children to a) make rubocop happy and b) make it's purpose clearer.

Is that ok?